pkg/routeagent/controllers/route/vxlan.go

```diff
@@ -0,0 +1,188 @@
+package route
+
+import (
+	"net"
+	"syscall"
+
+	"github.com/vishvananda/netlink"
+	"golang.org/x/sys/unix"
+	"k8s.io/klog"
+)
+
+type vxLanAttributes struct {
+	name     string
+	vxlanId  int
+	group    net.IP
+	srcAddr  net.IP
+	vtepPort int
+	mtu      int
+}
+
+type vxLanIface struct {
+	link *netlink.Vxlan
+}
+
+func newVxlanIface(attrs *vxLanAttributes) (*vxLanIface, error) {
+	iface := &netlink.Vxlan{
+		LinkAttrs: netlink.LinkAttrs{
+			Name:  attrs.name,
+			MTU:   attrs.mtu,
+			Flags: net.FlagUp,
+		},
+		VxlanId: attrs.vxlanId,
+		SrcAddr: attrs.srcAddr,
+		Group:   attrs.group,
+		Port:    attrs.vtepPort,
+	}
+
+	vxLANIface := &vxLanIface{
+		link: iface,
+	}
+
+	if err := createVxLanIface(vxLANIface); err != nil {
+		return nil, err
+	}
+
+	return vxLANIface, nil
+}
+
+func createVxLanIface(iface *vxLanIface) error {
+	err := netlink.LinkAdd(iface.link)
+	if err == syscall.EEXIST {
+		// Get the properties of existing vxlan interface
+		existing, err := netlink.LinkByName(iface.link.Name)
+		if err != nil {
+			return err
+		}
+
+		if isVxlanConfigTheSame(iface.link, existing) {
+			klog.V(4).Infof("VxLAN interface already exists with same configuration.")
+			iface.link = existing.(*netlink.Vxlan)
+			return nil
+		}
+
+		// Config does not match, delete the existing interface and re-create it.
+		if err = netlink.LinkDel(existing); err != nil {
+			klog.Errorf("Failed to delete the existing vxlan interface: %v", err)
+			return err
```

Comment:
As commented earlier and for consistency, can we return a formatted error instead of logging? Same elsewhere in this file.

Reply:
Done